api/v1alpha1/nodepool_types.go

```diff
@@ -81,6 +81,12 @@ type AWSNodePoolPlatform struct {
 	InstanceType    string                `json:"instanceType"`
 	InstanceProfile string                `json:"instanceProfile,omitempty"`
 	Subnet          *AWSResourceReference `json:"subnet,omitempty"`
+	// AMI is the image id to use
+	// +optional
+	AMI string `json:"ami,omitempty"`
+	// SecurityGroups is the set of security groups to associate with nodepool machines
+	// +optional
+	SecurityGroups []string `json:"securityGroups,omitempty"`
```

Comment:
what does this expect? and ID?
I think we should consolidate with `Subnet` and let both either support AWSResourceReference or only ID? Happy with this being done in a follow up though

Reply:
Yup, right now it's expecting an ID. Can definitely switch to AWSResourceReference